### PR TITLE
Subnet to VLAN Mapping

### PR DESCRIPTION
**Pull request by jdrew82**
This PR covers the FR in #87. One edge case that needs to be handled is the potential for more than one VLAN to be assigned to a Subnet. Nautobot doesn't allow for this but we can utilize Relationships to still provide that connection for the extra VLANs.

### CODE REVIEW

File: nautobot_ssot_infoblox/diffsync/adapters/infoblox.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def load_prefixes(self):
                 description=_pf.get("comment", ""),
                 status=_pf.get("status", "active"),
                 ext_attrs=get_ext_attr_dict(extattrs=_pf.get("extattrs", {})),
+                vlans=build_vlan_map(vlans=_pf["vlans"]) if _pf.get("vlans") else {},
```
Comment (chadell) [inline nautobot_ssot_infoblox/diffsync/adapters/infoblox.py:62]: `_pf["vlans"]` this looks to me like a potential `KeyError` candidate. You are actually handling it with a `get` in the second part

File: nautobot_ssot_infoblox/diffsync/adapters/infoblox.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ class InfobloxAdapter(DiffSync):
     vlangroup = InfobloxVLANView
     vlan = InfobloxVLAN
 
-    top_level = ["prefix", "ipaddress", "vlangroup", "vlan"]
+    top_level = ["vlangroup", "vlan", "prefix", "ipaddress"]
```
Comment (chadell) [inline nautobot_ssot_infoblox/diffsync/adapters/infoblox.py:27]: why the reordering?

File: nautobot_ssot_infoblox/diffsync/models/nautobot.py [hunk 3]
```diff
@@ -87,6 +90,39 @@ def create(cls, diffsync, ids, attrs):
             status=status,
             description=attrs.get("description", ""),
         )
+        if attrs.get("vlans") and len(attrs["vlans"].keys()) >= 1:
```
Comment (chadell) [inline nautobot_ssot_infoblox/diffsync/models/nautobot.py:93]: I would say that the `len(attrs["vlans"].keys()) >= 1` is redundant... If you get something that is not empty from "vlans" it's enough

File: nautobot_ssot_infoblox/diffsync/models/nautobot.py [hunk 4]
```diff
@@ -87,6 +90,39 @@ def create(cls, diffsync, ids, attrs):
...
+            relationship_dict = {
```
Comment (chadell) [inline nautobot_ssot_infoblox/diffsync/models/nautobot.py:94]: isn't this relationship good enough?
https://github.com/nautobot/nautobot/blob/develop/nautobot/ipam/models.py#L469

**Review by chadell — COMMENTED** [review]
some comments

File: nautobot_ssot_infoblox/diffsync/models/nautobot.py (hunk 3 above)
Comment (jdrew82) [reply nautobot_ssot_infoblox/diffsync/models/nautobot.py:93]: Good point

File: nautobot_ssot_infoblox/diffsync/models/nautobot.py (hunk 4 above)
Comment (jdrew82) [reply nautobot_ssot_infoblox/diffsync/models/nautobot.py:94]: The Relationship is for when there are more than one VLAN assigned to a Subnet. The only way to document that is with a Relarionship.

File: nautobot_ssot_infoblox/diffsync/adapters/infoblox.py (hunk 1 above)
Comment (chadell) [reply nautobot_ssot_infoblox/diffsync/adapters/infoblox.py:62]: you are right

**Review by chadell — APPROVED** [review]
LGTM

**Pull request merged**